applications/DG-Max/DG-Max.cpp

```diff
@@ -188,7 +187,7 @@ int main(int argc, char** argv) {
             points.emplace_back("S");
             points.emplace_back("R");
         }
-        BandstructureGNUPlot<DIM> output(path, points, structure, &result);
+        BandstructureGNUPlot<DIM> output(path, points, structure, &(*result));
```

Comment:
why not remove & and * instead? I guess you might have to cast then.

Reply:
The reason is that the plotter at the moment expects a raw pointer, as it should work both with and without results, and now that the results are in a smart pointer they need to be unpacked. This is indeed not a perfect situation, but I'm not sure what the best design is to improve it. So if you have any good suggestions, then please let me know.

Reply:
.get() returns the raw pointer